robottelo/manifests.py

```diff
@@ -17,160 +12,103 @@
-    closes the handle, and returns the path to temporary file.
-
+class ManifestCloner(object):
+    """Manifest clonning utility class."""
+    def __init__(self, template=None, signing_key=None):
+        self.template = template
+        self.signing_key = signing_key
+
+    def _download_manifest_info(self):
+        """Download and cache the manifest information."""
+        self.template = requests.get(settings.fake_manifest.url).content
+        self.signing_key = requests.get(settings.fake_manifest.key_url).content
+        self.signature = PKCS1_v1_5.new(RSA.importKey(self.signing_key))
+
+    def clone(self):
+        """Clones a RedHat-manifest file.
+
+        Change the consumer ``uuid`` and sign the new manifest with
+        signing key. The certificate for the key must be installed on the
+        candlepin server in order to accept uploading the cloned
+        manifest.
+
+        :return: A file-like object (``BytesIO`` on Python 3 and
+            ``StringIO`` on Python 2) with the contents of the cloned
+            manifest.
+        """
+        if self.signing_key is None or self.template is None:
+            self._download_manifest_info()
+
+        template_zip = zipfile.ZipFile(six.BytesIO(self.template))
+        consumer_export_zip = zipfile.ZipFile(
+            six.BytesIO(template_zip.read('consumer_export.zip')))
+
+        # Generate a new consumer_export.zip file changing the consumer
+        # uuid.
+        consumer_export = six.BytesIO()
+        with zipfile.ZipFile(consumer_export, 'w') as new_consumer_export_zip:
+            for name in consumer_export_zip.namelist():
+                if name == 'export/consumer.json':
+                    consumer_data = json.loads(
+                        consumer_export_zip.read(name).decode('utf-8'))
+                    consumer_data['uuid'] = six.text_type(uuid.uuid1())
+                    new_consumer_export_zip.writestr(
+                        name,
+                        json.dumps(consumer_data)
+                    )
+                else:
+                    new_consumer_export_zip.writestr(
+                        name,
+                        consumer_export_zip.read(name)
+                    )
+
+        # Generate a new manifest.zip file with the generated
+        # consumer_export.zip and new signature.
+        manifest = six.BytesIO()
+        with zipfile.ZipFile(
+                manifest, 'w', zipfile.ZIP_DEFLATED) as manifest_zip:
+            consumer_export.seek(0)
+            manifest_zip.writestr(
+                'consumer_export.zip',
+                consumer_export.read()
+            )
+            consumer_export.seek(0)
+            manifest_zip.writestr(
+                'signature',
+                self.signature.sign(SHA256.new(consumer_export.read()))
+            )
+        # Make sure that the file-like object is at the beginning and
+        # ready to be read.
+        manifest.seek(0)
```

Comment:
Why is this necessary? When you return a file-like object, is it necessary to "rewind" it?